packages/amplify_core/lib/src/types/auth/session/auth_user.dart

```diff
@@ -5,34 +5,23 @@ import 'package:amplify_core/amplify_core.dart';
 
 part 'auth_user.g.dart';
 
-@zAmplifyGenericSerializable
-class AuthUser<Details extends SignInDetails>
-    with AWSSerializable<Map<String, Object?>>, AWSDebuggable {
+@zAmplifySerializable
+class AuthUser with AWSSerializable<Map<String, Object?>>, AWSDebuggable {
   const AuthUser({
     required this.userId,
     required this.username,
-    required this.signInDetails,
```

Comment:
Q: why are we changing this?

Reply:
There are tradeoffs to having abstract types in data classes (SignInDetails). Generics are out. If this is included, a `fromJson` factory cannot be created since SignInDetails has no concrete type to call. I'll add it back to keep the signature the same and remove the `fromJson` factory.

Reply:
Actually, I'd rather keep it this way. I want to refine the type on `CognitoAuthUser` which means overriding the `signInDetails` field. Overriding a field with a field is a no-no, but overriding a getter with a field is okay.